oshi-core/src/main/java/oshi/software/os/unix/solaris/SolarisFileSystem.java

```diff
@@ -182,6 +179,20 @@ public long getMaxFileDescriptors() {
         return 0L;
     }
 
+    @Override
+    public long getMaxFileDescriptorsPerProcess() {
+        final List<String> lines = FileUtil.readFile("/etc/system");
+        for (final String line : lines) {
+            if (line.matches("set.+rlim_fd_cur=\\d+")) {
+                final String[] split = line.split("set.+rlim_fd_cur=");
```

Comment:
If you use a capturing regex above you can just extract the `group()` and not worry about the split and other checking.